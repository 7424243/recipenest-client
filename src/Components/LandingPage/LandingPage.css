/* Classes */

.demo-creds {
    font-family: 'Noto Serif', serif;
    text-align: left;
    margin-left: 20%;
    margin-right: 20%;
}

.landing-page-button {
    margin-top: 20px;
    background-color: #5F7466;
    color: white;
    font-size: 16px;
    padding: 10px 24px;
    font-family: 'Baskervville', serif;
    border-radius: 8px;
    border: none;
}

.landing-page-button:hover {
    box-shadow: 0 12px 16px 0 rgba(0,0,0,0.24), 0 17px 50px 0 rgba(0,0,0,0.19);
}

.landing-page-button-container {
    text-align: center;
}

.landing-page-container {
    line-height: 2.0;
    margin: 25px;
    font-family: 'Baskervville', serif;
    font-size: large;
}

.landing-page-info {
    text-align: left;
    margin-left: 20%;
    margin-right: 20%;
}

.landing-page-list {
    text-align: left;
}

.landing-page-li {
    margin-left: 20%;
    list-style-position: inside;
}


/* @media */

@media screen and (min-width: 540px) {
    .landing-page-container {
        margin: 100px 75px;
    }
}