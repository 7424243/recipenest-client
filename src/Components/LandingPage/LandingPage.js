import React from 'react'
import { Link } from 'react-router-dom'


function LandingPage() {
    return (
        <>
            <p>RecipeNest is a platform to help keep track of virtual recipes. It allows you to save links and notes for recipes that you have tried. </p>
            <p>Initially, you are shown a list of previously saved recipe names and a fun word cloud that was created based on the recipe entry's description. To view a saved recipe, click on the recipe name. To add a new recipe, click on 'Add Recipe' and you will be prompted to provide some details such as the name of the recipe, the link, a description, and any additional notes you want.</p>
            <Link to='/recipes'><button>Get Started!</button></Link>
        </>
    )
}

export default LandingPage