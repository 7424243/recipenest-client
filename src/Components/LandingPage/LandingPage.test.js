import React from 'react'
import ReactDOM from 'react-dom'
import {BrowserRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'
import LandingPage from './LandingPage';

describe('LandingPage Component', () => {

  //smoke test
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><LandingPage/></BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  //snapshot test
  it('renders the UI as expected', () => {
    const tree = renderer 
      .create(<BrowserRouter><LandingPage/></BrowserRouter>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  
})
