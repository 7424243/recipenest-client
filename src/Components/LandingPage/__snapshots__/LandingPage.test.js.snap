// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage Component renders the UI as expected 1`] = `
<div
  className="landing-page-container"
>
  <p
    className="landing-page-info"
  >
    RecipeNest is a platform created to help keep track of notes for virtual recipes.
  </p>
  <p
    className="landing-page-info"
  >
    Once you are logged in, the app allows you to...
  </p>
  <ul
    className="landing-page-list"
  >
    <li
      className="landing-page-li"
    >
      Access your own saved recipe notes.
    </li>
    <li
      className="landing-page-li"
    >
      Edit, delete, and add your own recipe notes.
    </li>
  </ul>
  <p
    className="landing-page-info"
  >
    Initially, you will be prompted to login - feel free to Sign Up to create your own account or use the demo credientials below. 
  </p>
  <p
    className="demo-creds"
  >
    Username: demo
  </p>
  <p
    className="demo-creds"
  >
    Password: Password0!
  </p>
  <p
    className="landing-page-info"
  >
    To view a saved recipe, simply click on the recipe.
  </p>
  <p
    className="landing-page-info"
  >
    To add a new recipe, click on the '+' button.
  </p>
  <br />
  <p
    className="landing-page-info"
  >
    If you choose to sign up for an account, keep in mind that you need to add recipe notes in order to see anything on the My Recipes page!!
  </p>
  <div
    className="landing-page-button-container"
  >
    <a
      href="/login"
      onClick={[Function]}
    >
      <button
        className="landing-page-button"
      >
        Get Started!
      </button>
    </a>
  </div>
</div>
`;
